lib/lambda_ethereum_consensus/store/blob_db.ex

```diff
@@ -55,8 +72,55 @@ defmodule LambdaEthereumConsensus.Store.BlobDb do
     end
   end
 
+  @spec prune_blobs_older_than(non_neg_integer()) :: :ok | {:error, String.t()} | :not_found
+  def prune_blobs_older_than(slot) do
+    Logger.info("[BlobDb] Pruning started.", slot: slot)
+    last_finalized_key = slot |> block_root_key(0)
+
+    with {:ok, it} <- Db.iterate(),
+         {:ok, @block_root_prefix <> _, _value} <-
+           Exleveldb.iterator_move(it, last_finalized_key),
+         {:ok, keys_to_remove} <- get_block_root_keys_to_remove(it),
+         :ok <- Exleveldb.iterator_close(it) do
+      keys_to_remove |> Enum.each(&remove_blob_by_block_root_key/1)
```

Comment:
We shouldn't do an enum.each and ignore the result of each removal. We should in the bare minimum log failures. If not, we can take care of it with a reduce.